cu29/src/monitoring.rs

```diff
@@ -122,3 +122,203 @@ impl Drop for ScopedAllocCounter {
         // );
     }
 }
+
+use hdrhistogram::Histogram;
+
+/// Accumulative stat object that can give your some real time statistics.
+pub struct LiveStatistics {
+    stats: Histogram<u64>, // u64 is the Counter type.
+}
+
+impl LiveStatistics {
+    pub fn new() -> Self {
+        LiveStatistics {
+            stats: Histogram::<u64>::new(3).unwrap(),
+        }
+    }
+
+    pub fn new_with_max(max: u64) -> Self {
+        LiveStatistics {
+            stats: Histogram::<u64>::new_with_bounds(1, max, 3).unwrap(),
+        }
+    }
+
+    #[inline]
+    pub fn min(&self) -> u64 {
+        self.stats.min()
+    }
+
+    #[inline]
+    pub fn max(&self) -> u64 {
+        self.stats.max()
+    }
+
+    #[inline]
+    pub fn mean(&self) -> f64 {
+        self.stats.mean()
+    }
+
+    #[inline]
+    pub fn percentile(&self, percentile: f64) -> u64 {
+        self.stats.value_at_quantile(percentile)
+    }
+
+    /// Adds a value to the statistics.
+    #[inline]
+    pub fn record(&mut self, value: u64) {
+        self.stats.record(value).unwrap();
```

Comment:
Does this panic if we hit the max record count?

Not sure if that's ideal, it could only keep a running average and max/min in that case

Reply:
indeed, that was a little lazy on my part this one :P